kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -130,7 +130,7 @@ object Async:
                     IO.Unsafe {
                         val sleepFiber = clock.unsafe.sleep(after)
                         val task       = IOTask[Ctx, E | Timeout, A](v, trace, context)
-                        sleepFiber.onComplete(_ => discard(task.complete(Result.fail(Timeout(frame)))))
+                        sleepFiber.onComplete(_ => discard(task.interrupt(Result.Fail(Timeout(frame)))))
```

Comment:
main fix is here: interrupt with the timeout failure instead of just completing the task